AudioKit/Platforms/Common/CsoundObj.m

```diff
@@ -89,41 +99,61 @@ - (instancetype)init
 
 - (void)sendScore:(NSString *)score
 {
+#ifdef TRAVIS_CI
+    [_scoMessages addObject:score];
+#else
     if (_cs) {
         csoundInputMessage(_cs, (const char *)[score cStringUsingEncoding:NSASCIIStringEncoding]);
     }
+#endif
 }
 
 - (void)play:(NSString *)csdFilePath
 {
+#ifdef TRAVIS_CI
+#else
     self.shouldRecord = NO;
     self.thread = [[NSThread alloc] initWithTarget:self
                                           selector:@selector(runCsound:)
                                             object:csdFilePath];
     [self.thread start];
+#endif
 }
 
 - (void)updateOrchestra:(NSString *)orchestraString
 {
+#ifdef TRAVIS_CI
+    [_orcMessages addObject:orchestraString];
+#else
     if (_cs) {
         csoundCompileOrc(_cs, (const char *)[orchestraString cStringUsingEncoding:NSASCIIStringEncoding]);
     }
+#endif
 }
 
 - (void)stop {
+#ifdef TRAVIS_CI
+#else
```

Comment:
I'm not completely convinced you do need to comment out all this code for testing. Even if you're rendering faster than realtime, you should still want to finish cleanly.